packages/contextual-menu-react/src/ContextualMenu.tsx

```diff
@@ -142,6 +144,34 @@ const ContextualMenuComponent = forwardRef<HTMLButtonElement, ContextualMenuProp
         theme = parseInt(backgroundColor.replace("#", ""), 16) < 0xffffff / 2 ? "dark" : "light";
     }
 
+    const [density, setDensity] = useState<string | null>(null);
+
+    useEffect(() => {
+        // Vi har samme problemet med density som vi hadde med dark/light men løsnignen er ikke like enkel.
+        // Vi må rett og slett grave oss fram til om det finnes et sted over oss der density settes og passe
+        // på at vi holder oss i sync.
+        if (rootRef.current) {
+            let node = rootRef.current as HTMLElement;
+            let density: string | null = null;
+            while (node && !density) {
+                density = node.getAttribute?.("data-layout-density");
+                if (!density) {
+                    node = node.parentNode as HTMLElement;
+                }
+            }
+            setDensity(density);
+
+            const observer = new MutationObserver(() => {
+                setDensity(node.getAttribute("data-layout-density"));
+            });
+            if (density && node) {
+                observer.observe(node, { attributes: true, attributeFilter: ["data-layout-density"] });
+            }
+            return () => observer.disconnect();
+        }
+        return () => {};
+    }, [setDensity, rootRef.current?.getAttribute]);
```

Comment:
Kan vi ikke sette verdien av en CSS-variabel på triggeren ut fra om det er compact eller ikke, og så lese den av som vi gjør for farge? Så slipper vi å traversere DOMet utover og heller bare sjekke triggeren.

```scss
// Disse kan settes i stilarket til ContextualMenu, eller for så vidt i core
@include jkl.comfortable-density-variables {
    --jkl-density: 'comfortable';
}
@include jkl.compact-density-variables {
    --jkl-density: 'compact';
}
```

```tsx
let density: string | undefined;
if (refs.reference.current) {
    density = getComputedStyle(refs.reference.current as HTMLElement).getPropertyValue(
        "--jkl-density",
    );
}
```

Reply:
Ulempen med å lese den av i render er at man fort blir hengende litt etter når det skiftes mellom compact/comfortable i løsningen vår og det ikke plukkes opp at det er endret før neste render etter at det skjer.

Men jeg kan lage en kombinasjon hvor jeg bruker en mutation-observer på triggeren og leser både farge og css-variabelen derfra?

Reply:
Hva tenker du @piofinn ? Tror som sagt mutationobserver er en bedre løsning men er enig i at det nok holder å se på triggeren og ikke traversere oppover hele treet.